racetrack_commons/racetrack_commons/deploy/job_type.py

```diff
@@ -49,14 +49,30 @@ def gather_job_types(
     for plugin_job_types in plugin_results:
         if plugin_job_types:
             for job_full_name, job_data in plugin_job_types.items():
-                base_image_path, template_path = job_data
-                assert base_image_path.is_file(), f'cannot find base image Dockerfile for {job_full_name} language wrapper: {base_image_path}'
-                assert template_path.is_file(), f'cannot find Dockerfile template for {job_full_name} language wrapper: {template_path}'
+
+                if isinstance(job_data, tuple):
+                    base_image_paths = [job_data[0]]
+                    template_paths = [job_data[1]]
+                elif isinstance(job_data, list):
+                    base_image_paths = [item[0] for item in job_data]
+                    template_paths = [item[1] for item in job_data]
+                else:
+                    raise RuntimeError(f'Invalid job type data. It should be list[tuple[Path, Path]], was {type(job_data)}')
+
+                assert len(base_image_paths) > 0, 'Job type should have non-empty list of base'
+
+                for i, base_image_path in enumerate(base_image_paths):
+                    template_path = template_paths[i]
+                    if base_image_path is not None:
+                        assert base_image_path.is_file(), f'cannot find base image Dockerfile for {job_full_name} language wrapper: {base_image_path}'
+                    if template_path is not None:
+                        assert template_path.is_file(), f'cannot find Dockerfile template for {job_full_name} language wrapper: {template_path}'
+
```

Comment:
Nitpick: putting these to separate validation function would make for shorter/simpler main function code.